include/toml++/impl/path.inl

```diff
@@ -29,40 +29,122 @@ TOML_ENABLE_WARNINGS;
 TOML_NAMESPACE_START
 {
 	TOML_EXTERNAL_LINKAGE
-	path_component::path_component(size_t index) noexcept : value_(index), type_(path_component_type::array_index)
-	{ }
+	path_component::path_component() //
+		: type_{ path_component_type::key }
+	{
+		path_component::store_key("", value_storage_);
```

Comment:
All of these `store_XXXX` calls shouldn't need the explicit `path_component::` prefix - _"the calls are coming from inside the house"_

Reply:
Added to make explicit that these are static methods :-) Style stuff, will remove!